### PR TITLE
Update k8s dependencies to 1.10

### LINKED ISSUE #676: errors relabeling pods on kubernetes 1.10
Reported from the slack channel, we seem to get errors relabeling pods in K8s 1.10.  We should test on K8s 1.10 and update the client-go dependency if we need to.

### PR DESCRIPTION
**Pull request by life1347**
This PR aims to solve the relabeling problem describe in #676 . 

In newer dependencies, every CRD object is required to implement `DeepCopy` methods. In order to use code-generator provided by Kubernetes to generate `DeepCopy` as well as CRD client, we need to follow it's directory structure and move all CRD structs to `github.com/fission/fission/pkg/apis/fission.io/v1`. 

Since it's a huge change to the code base, in order to speed up code review I adopt a golang feature called `type alias` introduced in go1.9 for gradual code repair purpose. In future development, we should migrate the type import from `github.com/fission/fission` to `github.com/fission/fission/pkg/apis/fission.io/v1`.

Backward compatibility test on GKE
- [x] K8S 1.9 (1.9.7-gke.0)
- [x] K8S 1.8 (1.8.12-gke.0)

```
$ gcloud container clusters create k8s --enable-kubernetes-alpha --cluster-version <version>
```

---
This change is [Reviewable](https://reviewable.io/reviews/fission/fission/687)

### CODE REVIEW

File: pkg/apis/fission.io/v1/validation.go [hunk 1]
```diff
@@ -38,26 +38,28 @@ var (
...
-	// Example: FunctionReference.Name
-	Field string
+		// Name of error field.
+		// Example: FunctionReference.Name
+		Field string
 
-	// Error field value.
-	BadValue interface{}
+		// Error field value.
+		BadValue string
```
Comment (life1347) [inline pkg/apis/fission.io/v1/validation.go]: Change from the interface to string for code generation. Not found a better way to solve the code generation problem when field type is interface.

File: pkg/apis/fission.io/v1/structs.gotes [hunk 2]
```diff
@@ -0,0 +1,359 @@
+package v1
```
Comment (soamvasani) [inline pkg/apis/fission.io/v1/structs.gotes]: What's this file about?  Add some explanatory comments?

File: types.go [hunk 3]
```diff
@@ -18,296 +18,39 @@ package fission
 
 import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	apiv1 "k8s.io/client-go/pkg/api/v1"
+
+	fv1 "github.com/fission/fission/pkg/apis/fission.io/v1"
 )
 
 type (
-	//
```
Comment (soamvasani) [inline types.go]: Why does all this have to be moved?  Is it possible to leave these here and use them in the Kubernetes types as fission.* ?

**Comment by soamvasani** [comment]
BTW, I saw a similar error on Kubernetes 1.9 as well, I'm not exactly sure why.  So this change is definitely important and we should make sure it makes it into 0.8.0.

**Comment by soamvasani** [comment]
Reviewed 36 of 57 files at r1, 1 of 1 files at r2, 3 of 3 files at r3, 4 of 16 files at r4.
Review status: all files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[.travis.yml, line 12 at r4](https://reviewable.io/reviews/fission/fission/687#-LDDhSUF8hIVM860AGz8:-LDDhSUG7qJTysNCKjvB:bwgeq61) ([raw file](https://github.com/fission/fission/blob/e1fa46eddddcf9a13280d634e3695d01af89ffee/.travis.yml#L12)):*
> ```YAML
> 
> go:
>   - 1.9.6
> ```

why not 1.10?

---

*[crd/types.go, line 426 at r4](https://reviewable.io/reviews/fission/fission/687#-LDDiRIa1ejFqfyJo3v9:-LDDiRIa1ejFqfyJo3vA:b-5tv73b) ([raw file](https://github.com/fission/fission/blob/e1fa46eddddcf9a13280d634e3695d01af89ffee/crd/types.go#L426)):*
> ```Go
> //	}
> //	return result.ErrorOrNil()
> //}
> ```

please remove all the commented stuff

---

*[fission/portforward.go, line 116 at r4](https://reviewable.io/reviews/fission/fission/687#-LDDiviz8Ly0wiTPvqdP:-LDDivj-D1eL7UAgsRkS:bs7xdou) ([raw file](https://github.com/fission/fission/blob/e1fa46eddddcf9a13280d634e3695d01af89ffee/fission/portforward.go#L116)):*
> ```Go
> 	transport, upgrader, err := spdy.RoundTripperFor(config)
> 	if err != nil {
> 		msg := fmt.Sprintf("spdy round tripper errored out :%v", err.Error())
> ```

Can we make error something that the user can make sense of ... like "Failed to connect to Fission service on Kubernetes: ..."

---

*[pkg/apis/fission.io/v1/types.go, line 29 at r4](https://reviewable.io/reviews/fission/fission/687#-LDDjLR7Bx65RsPhwwR1:-LDDjLR7Bx65RsPhwwR2:b-iiss1y) ([raw file](https://github.com/fission/fission/blob/e1fa46eddddcf9a13280d634e3695d01af89ffee/pkg/apis/fission.io/v1/types.go#L29)):*
> ```Go
> 
> //
> // To add a Fission CRD type:
> ```

Is everything in this list still valid?  If there are more (or less) steps, could you update?

---


*Comments from [Reviewable](https://reviewable.io/reviews/fission/fission/687)*

**Comment by life1347** [comment]
Review status: 11 of 47 files reviewed at latest revision, 6 unresolved discussions.

---

*[pkg/apis/fission.io/v1/structs.gotes, line 1 at r2](https://reviewable.io/reviews/fission/fission/687#-LD13uk7R08A2N_JLs6l-r2-1:-LDHftOW90hSMR33ihfm:b-oicyv6) ([raw file](https://github.com/fission/fission/blob/886678f02adf498c8b5bdc09d81c0fccbf93cc5d/pkg/apis/fission.io/v1/structs.gotes#L1)):*
<details><summary><i>Previously, soamvasani (Soam Vasani) wrote…</i></summary><blockquote>

What's this file about?  Add some explanatory comments?
</blockquote></details>

removed it.

---


*Comments from [Reviewable](https://reviewable.io/reviews/fission/fission/687)*

**Comment by life1347** [comment]
Review status: 11 of 47 files reviewed at latest revision, 6 unresolved discussions.

---

*[.travis.yml, line 12 at r4](https://reviewable.io/reviews/fission/fission/687#-LDDhSUF8hIVM860AGz8:-LDHg3r82ons7NnXa55u:b-896fix) ([raw file](https://github.com/fission/fission/blob/e1fa46eddddcf9a13280d634e3695d01af89ffee/.travis.yml#L12)):*
<details><summary><i>Previously, soamvasani (Soam Vasani) wrote…</i></summary><blockquote>

why not 1.10?
</blockquote></details>

Done.

---

*[crd/types.go, line 426 at r4](https://reviewable.io/reviews/fission/fission/687#-LDDiRIa1ejFqfyJo3v9:-LDHg5ao0bV8S_2BZr_v:b-896fix) ([raw file](https://github.com/fission/fission/blob/e1fa46eddddcf9a13280d634e3695d01af89ffee/crd/types.go#L426)):*
<details><summary><i>Previously, soamvasani (Soam Vasani) wrote…</i></summary><blockquote>

please remove all the commented stuff
</blockquote></details>

Done.

---

*[fission/portforward.go, line 116 at r4](https://reviewable.io/reviews/fission/fission/687#-LDDiviz8Ly0wiTPvqdP:-LDHg6TR2nUHl66Sqmgm:b-896fix) ([raw file](https://github.com/fission/fission/blob/e1fa46eddddcf9a13280d634e3695d01af89ffee/fission/portforward.go#L116)):*
<details><summary><i>Previously, soamvasani (Soam Vasani) wrote…</i></summary><blockquote>

Can we make error something that the user can make sense of ... like "Failed to connect to Fission service on Kubernetes: ..."
</blockquote></details>

Done.

---

*[pkg/apis/fission.io/v1/types.go, line 29 at r4](https://reviewable.io/reviews/fission/fission/687#-LDDjLR7Bx65RsPhwwR1:-LDHg8ZC3VMUuhO4xHV7:b-896fix) ([raw file](https://github.com/fission/fission/blob/e1fa46eddddcf9a13280d634e3695d01af89ffee/pkg/apis/fission.io/v1/types.go#L29)):*
<details><summary><i>Previously, soamvasani (Soam Vasani) wrote…</i></summary><blockquote>

Is everything in this list still valid?  If there are more (or less) steps, could you update?
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io/reviews/fission/fission/687)*

**Comment by soamvasani** [comment]
LGTM

---

Review status: all files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io/reviews/fission/fission/687#-:-LDsX7YZACb1eLbwVu4D:bb6ywmx)*

**Pull request merged**